test/jdk/jdk/jfr/event/runtime/TestNativeLibraryLoadEvent.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (c) 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package jdk.jfr.event.runtime;
+
+import static jdk.test.lib.Asserts.assertTrue;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import jdk.jfr.Recording;
+import jdk.jfr.consumer.RecordedEvent;
+import jdk.test.lib.Platform;
+import jdk.test.lib.jfr.EventNames;
+import jdk.test.lib.jfr.Events;
+
+/**
+ * @test
+ * @bug 8313251
+ * @key jfr
+ * @requires vm.hasJFR
+ * @library /test/lib
+ * @run main/othervm jdk.jfr.event.runtime.TestNativeLibraryLoadEvent
+ */
+public class TestNativeLibraryLoadEvent {
+
+    private final static String EVENT_NAME = EventNames.NativeLibraryLoad;
+
+    public static void main(String[] args) throws Throwable {
+        try (Recording recording = new Recording()) {
+            recording.enable(EVENT_NAME);
+            recording.start();
+            System.loadLibrary("awt");
+            recording.stop();
+
+            List<String> expectedLibs = getExpectedLibs();
+            for (RecordedEvent event : Events.fromRecording(recording)) {
+                System.out.println("Event:" + event);
+                String lib = Events.assertField(event, "name").notEmpty().getValue();
+                Events.assertField(event, "success");
+                for (String expectedLib : new ArrayList<>(expectedLibs)) {
+                    if (lib.contains(expectedLib)) {
+                        expectedLibs.remove(expectedLib);
+                    }
+                }
+            }
+            assertTrue(expectedLibs.isEmpty(), "Missing libraries:" + expectedLibs.stream().collect(Collectors.joining(", ")));
+        }
+    }
+
+    private static List<String> getExpectedLibs() throws Throwable {
+        String libTemplate = null;
+        if (Platform.isWindows()) {
+            libTemplate = "%s.dll";
+        } else if (Platform.isOSX()) {
+            libTemplate = "lib%s.dylib";
```

Comment:
We have `Platform.sharedLibraryExt()` to simplify this a bit. Unfortunately we don't have a `sharedLibraryPrefix()` to reduce it to one line. :(

Reply:
Okay

Reply:
I created  https://bugs.openjdk.org/browse/JDK-8313670  .